src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialChisel.java

```diff
@@ -0,0 +1,438 @@
+			int aEUPercent = getEuDiscountForParallelism();
+			int aSpeedBonusPercent = 200;
+
+			aMaxParallelRecipes = this.canBufferOutputs(tRecipe, aMaxParallelRecipes);
+			if (aMaxParallelRecipes == 0) {
+				log("BAD RETURN - 2");
+				return false;
+			}
+
+			// EU discount
+			float tRecipeEUt = (tRecipe.mEUt * aEUPercent) / 100.0f;
+			float tTotalEUt = 0.0f;
+
+			int parallelRecipes = 0;
+
+			log("parallelRecipes: "+parallelRecipes);
+			log("aMaxParallelRecipes: "+aMaxParallelRecipes);
+			log("tTotalEUt: "+tTotalEUt);
+			log("tVoltage: "+tVoltage);
+			log("tRecipeEUt: "+tRecipeEUt);
+			// Count recipes to do in parallel, consuming input items and fluids and considering input voltage limits
+			for (; parallelRecipes < aMaxParallelRecipes && tTotalEUt < (tEnergy - tRecipeEUt); parallelRecipes++) {
+				if (!tRecipe.isRecipeInputEqual(true, aFluidInputs, aItemInputs)) {
+					log("Broke at "+parallelRecipes+".");
+					break;
+				}
+				log("Bumped EU from "+tTotalEUt+" to "+(tTotalEUt+tRecipeEUt)+".");
+				tTotalEUt += tRecipeEUt;
+			}
+
+			if (parallelRecipes == 0) {
+				log("BAD RETURN - 3");
+				return false;
+			}
+
+			// -- Try not to fail after this point - inputs have already been consumed! --
+			log("parallelRecipes: "+parallelRecipes);
+			log("aMaxParallelRecipes: "+aMaxParallelRecipes);
+			log("tTotalEUt: "+tTotalEUt);
+			log("tVoltage: "+tVoltage);
+			log("tRecipeEUt: "+tRecipeEUt);
+
+
+			// Convert speed bonus to duration multiplier
+			// e.g. 100% speed bonus = 200% speed = 100%/200% = 50% recipe duration.
+			aSpeedBonusPercent = Math.max(-99, aSpeedBonusPercent);
+			float tTimeFactor = 100.0f / (100.0f + aSpeedBonusPercent);
+			this.mMaxProgresstime = (int)(tRecipe.mDuration * tTimeFactor);
+
+			this.mEUt = (int)Math.ceil(tTotalEUt);
+
+			this.mEfficiency = (10000 - (getIdealStatus() - getRepairStatus()) * 1000);
+			this.mEfficiencyIncrease = 10000;		
+
+			// Overclock
+			if (this.mEUt <= 16) {
+				this.mEUt = (this.mEUt * (1 << tTier - 1) * (1 << tTier - 1));
+				this.mMaxProgresstime = (this.mMaxProgresstime / (1 << tTier - 1));
+			} else {
+				while (this.mEUt <= gregtech.api.enums.GT_Values.V[(tTier - 1)]) {
+					this.mEUt *= 4;
+					this.mMaxProgresstime /= 2;
+				}
+			}
+
+			if (this.mEUt > 0) {
+				this.mEUt = (-this.mEUt);
+			}
+
+			this.mMaxProgresstime = Math.max(1, this.mMaxProgresstime);
+
+			// Collect fluid outputs
+			FluidStack[] tOutputFluids = new FluidStack[tRecipe.mFluidOutputs.length];
+			for (int h = 0; h < tRecipe.mFluidOutputs.length; h++) {
+				if (tRecipe.getFluidOutput(h) != null) {
+					tOutputFluids[h] = tRecipe.getFluidOutput(h).copy();
+					tOutputFluids[h].amount *= parallelRecipes;
+				}
+			}
+
+			// Collect output item types
+			ItemStack[] tOutputItems = new ItemStack[tRecipe.mOutputs.length];
+			for (int h = 0; h < tRecipe.mOutputs.length; h++) {
+				if (tRecipe.getOutput(h) != null) {
+					tOutputItems[h] = tRecipe.getOutput(h).copy();
```

Comment:
Can probably be refactored into 
```java
int totalChance = tRecipe.getOutputChance(h) * parallelRecipes;
int stackSize = totalChance / 10000 * tRecipe.mOutputs[h].stackSize;
if (getBaseMetaTileEntity().getRandomNumber(10000) <= totalChance % 10000)
    stackSize += tRecipe.mOutputs[h].stackSize;
if (stackSize != 0) {
    tOutputItems[h] = tRecipe.getOutput(h).copy();
    tOutputItems[h].stackSize = stackSize;
}
```
and yeet the `Set output item stack sizes` loop and `Strip empty stacks` loop below